### PR TITLE
Add the endpoint /api/metrics for Prometheus.

### PR DESCRIPTION
**Pull request by eryshev**
Add the endpoint /api/metrics for Prometheus.
Delete a logging package.
Small refactoring in Executor.scala.

### CODE REVIEW

File: core/src/main/scala/com/criteo/cuttle/Metrics.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.criteo.cuttle
+
+sealed trait Metric {
+  def toString: String
+}
+
+case class Gauge(name: String, value: Long, tags: Seq[(String, String)] = Seq.empty) extends Metric
+
+trait MetricProvider {
+  private[cuttle] def getMetrics(jobs: Set[String]): Seq[Metric]
```
Comment (guillaumebort) [inline core/src/main/scala/com/criteo/cuttle/Metrics.scala]: Can't be `private[cuttle]` if it is exposed as part of the Scheduler API.

File: core/src/main/scala/com/criteo/cuttle/Metrics.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.criteo.cuttle
```
Comment (guillaumebort) [inline core/src/main/scala/com/criteo/cuttle/Metrics.scala]: Can we scope all this into a `Metrics` object, so in the public API it's all coherent?

File: core/src/main/scala/com/criteo/cuttle/Metrics.scala [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  def toString: String
+}
+
+case class Gauge(name: String, value: Long, tags: Seq[(String, String)] = Seq.empty) extends Metric
+
+trait MetricProvider {
+  private[cuttle] def getMetrics(jobs: Set[String]): Seq[Metric]
+}
+
+object Prometheus {
```
Comment (guillaumebort) [inline core/src/main/scala/com/criteo/cuttle/Metrics.scala]: This one should be `private[cuttle]` as it is an implementation detail.

**Comment by eryshev** [comment]
Corrected

File: core/src/main/scala/com/criteo/cuttle/Executor.scala [hunk 4]
```diff
@@ -378,6 +377,32 @@ class Executor[S <: Scheduling] private[cuttle] (
                                          limit: Int): Seq[ExecutionLog] =
     queries.getExecutionLog(queryContexts, jobs, sort, asc, offset, limit).transact(xa).unsafePerformIO
 
+  private[cuttle] def getStats(jobs: Set[String]): Json = {
+    val (running, waiting) = runningExecutionsSizes(jobs)
```
Comment (dufrannea) [inline core/src/main/scala/com/criteo/cuttle/Executor.scala]: I just wanted to point out that provided figures can be inconsistent because stm accesses are non enclosed in a single 
transaction. Not use this is critical though

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/intervals/IntervalMap.scala [hunk 5]
```diff
@@ -133,6 +133,10 @@ private[timeseries] object IntervalMap {
         m <- this.intersect(interval).toList
       } yield m): _*))
     }
+
```
Comment (dufrannea) [inline timeseries/src/main/scala/com/criteo/cuttle/timeseries/intervals/IntervalMap.scala]: You are adding code you do not use, If you really feel this is something that needs to be done I would suggest you could move it to another PR.

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/intervals/IntervalMap.scala [hunk 6]
```diff
@@ -162,4 +166,6 @@ private[timeseries] sealed trait IntervalMap[A, B] {
   def mapKeys[K: Ordering](f: A => K): IntervalMap[K, B]
   def whenIsDef[C](other: IntervalMap[A, C]): IntervalMap[A, B]
   def whenIsUndef[C](other: IntervalMap[A, C]): IntervalMap[A, B]
+  def head: (Interval[A], B)
```
Comment (dufrannea) [inline timeseries/src/main/scala/com/criteo/cuttle/timeseries/intervals/IntervalMap.scala]: Same here

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/intervals/IntervalMap.scala (hunk 5 above)
Comment (eryshev) [reply timeseries/src/main/scala/com/criteo/cuttle/timeseries/intervals/IntervalMap.scala]: Cleaned it

**Comment by eryshev** [comment]
@dufrannea All done
